apple/internal/multi_arch_binary_support.bzl

```diff
@@ -51,26 +53,39 @@ def subtract_linking_contexts(owner, linking_contexts, avoid_dep_linking_context
     libraries = []
     user_link_flags = []
     additional_inputs = []
+    non_library_inputs = []
     linkstamps = []
     avoid_library_set = _build_avoid_library_set(avoid_dep_linking_contexts)
+
     for linking_context in linking_contexts:
         for linker_input in linking_context.linker_inputs.to_list():
+            if not linker_input.libraries:
```

Comment:
im still not sure if this is nicer, or your version which just dedup's linker_inputs all together, which feels like it could lead to a lot less work in this loop / duplicate flags for non-avoided libraries

Reply:
I can go either way. My version is more aggressive, but it makes intuitive sense to me

Reply:
Per [this thread](https://bazelbuild.slack.com/archives/C04DFUBQGSU/p1744825677535229), I've put forward a patch using my more aggressive deduplication of whole linker inputs. I would like any reviewers to test the patch out in their own repo before approving, using either Bazel 7 or 8, to help ensure minimal breakage.